src/emulator/eventarcEmulator.ts

```diff
@@ -98,52 +113,62 @@ export class EventarcEmulator implements EmulatorInstance {
     const hub = express();
     hub.post([registerTriggerRoute], dataMiddleware, registerTriggerHandler);
     hub.post([publishEventsRoute], dataMiddleware, publishEventsHandler);
+    hub.post(
+      [publishNativeEventsRoute],
+      dataMiddleware,
+      cors({ origin: true }),
+      publishEventsHandler,
+    );
+    hub.get([getTriggersRoute], cors({ origin: true }), getTriggersHandler);
     hub.all("*", (req, res) => {
       this.logger.log("DEBUG", `Eventarc emulator received unknown request at path ${req.path}`);
       res.sendStatus(404);
     });
     return hub;
   }
 
-  async triggerCustomEventFunction(channel: string, event: CloudEvent<any>) {
+  async triggerEventFunction(channel: string, event: CloudEvent<any>): Promise<void[]> {
     if (!EmulatorRegistry.isRunning(Emulators.FUNCTIONS)) {
       this.logger.log("INFO", "Functions emulator not found. This should not happen.");
       return Promise.reject();
     }
     const key = `${event.type}-${channel}`;
-    const triggers = this.customEvents[key] || [];
+    const triggers = this.events[key] || [];
+    const eventPayload = channel === GOOGLE_CHANNEL ? event : cloudEventFromProtoToJson(event);
     return await Promise.all(
       triggers
         .filter(
           (trigger) =>
             !trigger.eventTrigger.eventFilters ||
             this.matchesAll(event, trigger.eventTrigger.eventFilters),
         )
-        .map((trigger) =>
-          EmulatorRegistry.client(Emulators.FUNCTIONS)
-            .request<CloudEvent<any>, NodeJS.ReadableStream>({
-              method: "POST",
-              path: `/functions/projects/${trigger.projectId}/triggers/${trigger.triggerName}`,
-              body: JSON.stringify(cloudEventFromProtoToJson(event)),
-              responseType: "stream",
-              resolveOnHTTPError: true,
-            })
-            .then((res) => {
-              // Since the response type is a stream and using `resolveOnHTTPError: true`, we check status manually.
-              if (res.status >= 400) {
-                throw new FirebaseError(`Received non-200 status code: ${res.status}`);
-              }
-            })
-            .catch((err) => {
-              this.logger.log(
-                "ERROR",
-                `Failed to trigger Functions emulator for ${trigger.triggerName}: ${err}`,
-              );
-            }),
-        ),
+        .map((trigger) => this.callFunctionTrigger(trigger, eventPayload)),
     );
   }
 
+  callFunctionTrigger(trigger: EmulatedEventTrigger, event: CloudEvent<any>): Promise<void> {
+    return EmulatorRegistry.client(Emulators.FUNCTIONS)
+      .request<CloudEvent<any>, NodeJS.ReadableStream>({
+        method: "POST",
+        path: `/functions/projects/${trigger.projectId}/triggers/${trigger.triggerName}`,
+        body: JSON.stringify(event),
+        responseType: "stream",
```

Comment:
q: why is the response type here stream?

Reply:
I am not 100% of this either. I know it is showing up as a diff here, but that's just because I extracted out some of the previous code into a function. I am inclined to leave it this way as it is working fine with supporting the new events and I don't want to break anything that was working previously.